src/Analyzer.Utilities/CSharp/ITypeSymbolExtensions.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace Analyzer.Utilities
+{
+    public static class ITypeSymbolExtensions
+    {
+        public static bool IsAssignableTo(this ITypeSymbol fromSymbol, ITypeSymbol toSymbol, Compilation compilation)
```

Comment:
but this still require explicit link to this dll from csharp specific project. once we have things like this, people would want to just create analyzer in common layer and delegate all language specific logic to this dll. like how code fixer does it. can we use things like MEF to do dependency injection?